tests/test_detection_pytest.py

```diff
@@ -0,0 +1,129 @@
+from btwa_api.app.detection import get_score, detect
+from datetime import datetime, timedelta
+
+
+def test_get_score():
+    # if signal <= 70:
+    assert get_score(0) == 35
+    assert get_score(50) == 35
+    assert get_score(70) == 35
+
+    # if signal <= 75:
+    assert get_score(74) == 25
+    assert get_score(75) == 25
+
+    # if signal <= 80:
+    assert get_score(76) == 17
+    assert get_score(80) == 17
+
+    # if signal <= 85:
+    assert get_score(84) == 10
+    assert get_score(85) == 10
+
+    # if signal <= 90:
+    assert get_score(88) == 5
+    assert get_score(90) == 5
+
+    # else
+    assert get_score(91) == 0
+
+    assert get_score(1022261615165151151) == 0
+    assert get_score(----1022261615165151151) == 0
+    assert get_score(1022261615165151151556565656565665565656) == 0
+
+
+def test_detect():
+    NO_OF_DAYS_TO_COUNT = 14
+
+    ONE_DAY = 1
+    TWO_DAY = 2
+    THREE_DAY = 3
+    FOR_DAY = 4
+    THIRTEEN_DAY = 13
+    FIFTEEN_DAY = 15
+
+    date_n_days_ago = datetime.now() - timedelta(days=ONE_DAY)
+    timestamp_ms_n_one_day_ago = date_n_days_ago.timestamp() * 1000
+    date_n_days_ago2 = datetime.now() - timedelta(days=TWO_DAY)
+    timestamp_ms_n_one_day_ago2 = date_n_days_ago2.timestamp() * 1000
+    date_no_days_to_count = datetime.now() - timedelta(days=FIFTEEN_DAY)
+    timestamp_date_no_days_to_count = date_no_days_to_count.timestamp() * 1000
+
+    # bad data, timestamp is in the past
```

Comment:
Nitpick: It could be rewritten to a table tests structure:
```
tests = [
{"input": ..., "want": ...},
{"input": ..., "want": ...},
{"input": ..., "want": ...},
]
```

and then iterate over `tests`. It'd add readability, but might be only subjective to me.

Reply:
term *expected* is typicaly used in this context

Reply:
IMO there should be a separate test function for each case, then you instantly know where is the problem if something breaks. If you iterate over data like this it takes some time to find for which inputs the test failed. In such case parametrization can be used (https://docs.pytest.org/en/latest/parametrize.html), which achieves what you describe and it add discoverability because it names the test after the inputs.

But here I don't really mind, whatever works :-)

Reply:
@Kobzol, ok, my bad. I'm used from Go which has `testing.Run()` method, which each iteration runs as a separate test and you know what broke.